### PR TITLE
Full Docker support

### PR DESCRIPTION
**Pull request by gaby**
The purpose of this draft PR is to track adding support for Docker as requested by various users on Reddit.

- [x] Add Docker Ignore
- [x] Add Dockerfile with Multi-Stage support
- [x] Update README with instructions on how to use Docker Image
- [x] Provide example `docker-compose.yml` for running StoreDown
- [x] Add support for building, tagging, and releasing `StoreDown` images to `ghcr.io` and possibly `DockerHub`

@FoxUSA Do you currently have an account for StoreDown in `hub.docker.com` ? If not, can you create one under `storedown` username. That way the image can be created as  `storedown/storedown:latest`

### CODE REVIEW

**Comment by FoxUSA** [comment]
Ahh fine.
I thought folks may want to just plop this on a web server of their choice .
I have supported [Docker](https://github.com/FoxUSA/OpenNote-Docker) and provided [compose instruction on other project in the past](https://github.com/FoxUSA/OpenNote/blob/master/docs/Install.md).

This app follows that same pattern.

**Comment by gaby** [comment]
From what I can see it's quite easy to download your artifacts and run them in a web server, but a lot of people prefer to just `docker run` an existing image to facilitate automation, etc. 

I'm working on PR #2 first, since that one seems crucial before starting to build/publish Docker Images.

**Comment by gaby** [comment]
@FoxUSA PR is ready

Final Docker image is 27.3 MB served using NGINX

Screenshots:

image

image

image

**Comment by gaby** [comment]
This PR is also under MIT license. I'd be adding changes to the readme in like 1hr. Which is the last thing todo.

**Comment by gaby** [comment]
* Updated README with instructions for running the Docker container
* Included an example docker-compose file for running StoreDown.

**Comment by gaby** [comment]
@FoxUSA where is the data from the Docker Container stored? So I can add it to the example.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# Build tools
+RUN apk add --no-cache \
+    build-base \
+    gcc \
+    g++ \
+    make
+
+# Install dependencies
+COPY package*.json ./
+RUN npm install
```
Comment (reeseovine) [inline Dockerfile:15]: You should use `npm ci` instead which is much better suited to Dockerfiles. It doesn't update package-lock.json unlike `npm install`.

Comment (gaby) [reply Dockerfile]: @reeseovine Done, thanks for the suggestion!

**Comment by gaby** [comment]
@FoxUSA I'm confused, did you rugpull my PR by merging a Dockerfile into master?

**Comment by gaby** [comment]
👍🏻 closing this

**Comment by FoxUSA** [comment]
- [ ] I had questions about the GitHubs registry, the reason for the NGINX config. 
- [ ] Then I need to just resolve the minor conflicts.

**Pull request closed without merging**